firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java

```diff
@@ -243,17 +244,28 @@ public void useFunctionsEmulator(String origin) {
       request = request.header("Firebase-Instance-ID-Token", context.getInstanceIdToken());
     }
 
+    OkHttpClient callClient = options.apply(client);
```

Comment:
Question: Do we create a client and apply these options to create a new Client for some reason?
I am wondering whether we can reduce state by just saying options.apply(new OKHttpClient()) or possibly options.createClient()

Reply:
The way `options.apply()` is implemented is the recommended way to do it. To change the timeout, you have to create a new client from the existing client, but they share most of their state, so it's not as expensive as it looks. The okhttp API is admittedly not obvious.

https://github.com/square/okhttp/wiki/Recipes#per-call-configuration

Reply:
As to why we don't just create a new one every time: doing it this way lets us share one connection pool and dispatcher across all of the function references for an app.

Reply:
Thanks for clarifying